api/base/exceptions.py

```diff
@@ -1,5 +1,35 @@
-from rest_framework.exceptions import APIException
+
 from rest_framework import status
+from rest_framework.exceptions import APIException
+
+
+def json_api_exception_handler(exc, context):
+    """ Custom exception handler that returns errors object as an array """
+
+    from rest_framework.views import exception_handler
+    response = exception_handler(exc, context)
+
+    # Title removed to avoid clash with node "title" errors
+    acceptable_members = ['id', 'links', 'status', 'code', 'detail', 'source', 'meta']
+    errors = []
+
+    if response is not None:
+        message = response.data
+        if isinstance(message, dict):
+            for key, value in message.iteritems():
+                if key in acceptable_members:
+                    errors.append({key: value})
+                else:
+                    errors.append({'detail': {key: value}})
```

Comment:
I might misunderstand your comment, but L23 is tested here:
https://github.com/pattisdr/osf.io/blob/feature/JSONAPI_responses/tests/api_tests/nodes/test_views.py#L1698

If you attempt to create a node in the API without providing a title, it tries to return a dictionary:
![screen shot 2015-08-26 at 2 20 32 pm](https://cloud.githubusercontent.com/assets/9755598/9502115/413388f0-4bfd-11e5-8970-6f2ef9938bb8.png)

Since `title` is not one of the acceptable top-level error keys (well it *is* but here it's referring to the blank field), the dict is nested under detail, as confirmed by the test:

![screen shot 2015-08-26 at 2 16 43 pm](https://cloud.githubusercontent.com/assets/9755598/9502153/823b54c2-4bfd-11e5-9d22-abfdefb9ddc0.png)